### ISSUE TITLE
Duplicated payload when using an asterisk

### ISSUE THREAD

**Issue by TechMagister** [drujensen/delimiter_tree#3]
Can be reproduced using this test : 
```crystal
  it "should not duplicate payload" do
    tree = Delimiter::Tree(Symbol).new
    tree.add "/*", :all
    tree.find("/test/part/hop").payload.should eq [:all] # FAIL here
  end
```
Got : 
```
1) Delimiter::Tree(T) should not duplicate payload
     Failure/Error: tree.find("/test/part/hop").payload.should eq [:all]

       expected: [:all]
            got: [:all, :all, :all]
```

Is ``` [:all, :all, :all] ``` the expected result ?
Because delimiter_tree is used in kemalyst and I have some troubles due to the asterisk in the routes. ( in this case an infinite loop, unless I modify the sources to uniq! the returned array ( is that a correct English sentence :scream: ?) )

**Comment by drujensen** [comment]
ugh, that's no good.  It's adding it for each delimiter found.  I will take a look when I get a chance.

**Comment by drujensen** [comment]
@TechMagister I've released a fix v0.1.3.  thanks for reporting this.
